### ISSUE TITLE
[Event Request] "Price List Line" - SetNextLineNo seed

### ISSUE THREAD

**Issue by fvet** [microsoft/ALAppExtensions#18974]
Looking at the following "Price list line" parameters:

- GetMaxPriceLineNo  : 1.000.000.000
- SetNextLineNo : 10.000
```
procedure SetNextLineNo()
    var
        PriceListLine: Record "Price List Line";
    begin
        "Line No." := 10000;
        PriceListLine.SetRange("Price List Code", "Price List Code");
        if PriceListLine.FindLast() then
            "Line No." += PriceListLine."Line No.";
    end;
```

we seems to be limited to **max. 100.000** price list lines per price list.
Even if we clean all historic price list lines, we would not be able to add new lines if the last / highest number has been used.

The reasoning will probably be found in the `AutoSplitKey` used on the price list line pages.

However, for vendors having large assortments (e.g. 10.000 price list lines is not an exception), we would import the price list lines (via the price list worksheet) and using 'Implement price changes', new price list lines would be created. 

Since we maintain **only one 'price list header' per vendor** and add new periods (e.g. per quarter) as price list lines within the **same** header, I'm afraid we will hit the limit / highest line no. rather soon.

We think it should be possible to increase (automatically created) price list lines (via the price worksheet lines) not per 10.000, but e.g. per 1 (we're aware of the impact on the 'AutoSplitKey'), either by adding a `OnBeforeSetNextLineNo` publisher, either by moving the 10.000 value to a separate `GetLineNoSeed` function with publisher.

Alternative would be to be able to use another seed / line no when using the 'Implement Price Changes' in the price worksheet.

```
// codeunit 7017 "Price List Management"

 local procedure ImplementNewPrice(var PriceWorksheetLine: Record "Price Worksheet Line"; var PriceListLine: Record "Price List Line"; var InsertedUpdatedLeft: array[3] of Integer) Implemented: Boolean;
    begin
        Implemented := false;
        PriceListLine.TransferFields(PriceWorksheetLine);
        if PriceWorksheetLine."Existing Line" then begin
            PriceListLine.Status := PriceListLine.Status::Draft;
            if PriceListLine.Modify(true) then begin
                InsertedUpdatedLeft[2] += 1;
                Implemented := true;
            end;
        end else begin
            PriceListLine.SetNextLineNo(); // <<<<  Alternative would be to be able to set another seed / line no here 
            if PriceListLine.Insert(true) then begin
                InsertedUpdatedLeft[1] += 1;
                Implemented := true;
            end;
        end;
        if Implemented then begin
            PriceWorksheetLine.Delete();
            PriceListLine.Mark(true);
        end;
        OnAfterImplementNewPrice(PriceWorksheetLine, PriceListLine, Implemented);
    end;
```

I'll let the liberty to the pricing list owning team to decide om how to best implement a fix.

---

- [ ] Reviewed / Found in latest BC version

**Comment by JakovljevicDusan** [comment]
Workaround provided with event OnBeforeSetNextLineNo() implemented for issue #21474.

**Comment by JakovljevicDusan** [comment]
Hi @fvet,

This is old request and I guess you probably already use event mentioned in the previous comment. 
There were probably different solutions, but anyway that event is there now so I am closing this request. 
Reasons:
 1. that function is tiny, so it doesn't make sense to overcomplicate extensibility
 2. we also don't want to introduce breaking changes when it is not really necessary.
